test/Microsoft.OpenAPI.OData.Reader.Tests/Resources/TripService.OpenApi.V2.json

```diff
@@ -25794,9 +25792,7 @@
         "Gender": {
           "$ref": "#/definitions/Microsoft.OData.Service.Sample.TrippinInMemory.Models.PersonGender"
         },
-        "Age": {
-          "format": "int64"
-        },
+        "Age": { },
```

Comment:
now I understand why the format was set on the root object. for v2....
the serialize as v2 probably needs some further fixing to make sure the int64 and double formats are present.

Reply:
Could this be a potential challenge since v2 doesn't support `anyOf` and `oneOf `constructs?

Reply:
To be specific: I'm not saying we should have anyof/oneof in v2 as you mentioned it's not supported. What I'm requesting is to have the type/format set on the root schema (extracted from the any/oneof at serialization time). Because with the current code, we're losing information over the previous version.

Reply:
Do we still want to retain the `Format` properties defined within the respective schemas of the `oneOf`?

Reply:
in case of v2 only we want to retain the format and type from one of the oneof/anyof. In case of v3, since we can have a more precise description, the root node shouldn't have any format/type, but the sub-schemas will contain the information. I hope that clarifies the point.

Reply:
Right, but we still don't want the format to appear on the root schema object when serializing to v3, only for v2

Reply:
Seems like this should be resolved in the OpenAPI.NET lib. I don't see a way to include the `Format` property in the root of the schema object for v2 and exclude it for v3.

Reply:
right, would you mind opening a PR on the openapi.net lib to tackle the problem end to end please?

Reply:
Here's the issue linked to that effect: https://github.com/microsoft/OpenAPI.NET/issues/992
Shall we proceed to merge this PR in the meantime, or is this fix required before proceeding?

Reply:
Thanks for creating the issue. I believe we should fix the OpenAPI.net library first since otherwise we'll ship a regression. I'll add my thoughts on the issue over there.

Reply:
preview is released, can you give it a try? https://www.nuget.org/packages/Microsoft.OpenApi/1.4.0-preview3